vowpalwabbit/parse_regressor.cc

```diff
@@ -250,28 +252,19 @@ void save_load_header(
         {
           char pair[3] = {0, 0, 0};
 
-          if (!read)
-          {
-            memcpy(pair, all.pairs[i].data(), 2);
-            // Copies data to stringstream regardless of existence of null characters.
-            // This might result in unintuitive behavior i.e. copy data after nulls as well.
-            msg.write(reinterpret_cast<char*>(all.pairs[i].data()), sizeof(all.pairs[i]));
-            msg << " ";
-          }
-
+          // Only the read path is implemented since this is for old version read support.
           bytes_read_write += bin_text_read_write_fixed_validated(model_file, pair, 2, "", read, msg, text);
-          if (read)
+          std::vector<namespace_index> temp(pair, pair+std::strlen(pair));
```

Comment:
do
```
char pair[2] = {0, 0};
...
std::vector<namespace_index> temp(pair, *(&pair + 1));
```
instead of the arbitrary strlen?

Reply:
https://www.geeksforgeeks.org/how-to-find-size-of-array-in-cc-without-using-sizeof-operator/
For an explanation of what `*(&pair + 1)` is actually doing.
Basically `&pair` is giving you an address of a `char[2]`. +1 increments the pointer by `sizeof(char[2])` (ie: increments the pointer by 2 bytes), and dereferencing that gives you a `char*` to 1 byte past the end of the initial array.

This is functionally equivalent to `sizeof(pair)/sizeof(pair[0])`